### ISSUE TITLE
Specify Interactive Mode in rclone.conf

### ISSUE THREAD

**Issue by jeff-geoff** [rclone/rclone#3886]
#### What is your current rclone version (output from `rclone version`)?

rclone v1.50.2
- os/arch: darwin/amd64
- go version: go1.13.4

#### What problem are you are trying to solve?
Accidental Deletion

#### How do you think rclone should be changed to solve that?

Add the ability to specify something like interactive-mode = true within rclone.conf

Enabling interactive mode would cause rclone to present Yes/No options to a user when conducting an operation that results in overwriting or deleting a file to prevent accidental data loss.

**Comment by ncw** [comment]
The mode for this would be something like this from the `cp` man page

       -i, --interactive
              prompt before delete or overwrite

Some careful locking will be required given that rclone is concurrent but there is only one stdin/stdout for talking to the user.

Finding all the uses of `DryRun` and turning them into a function something like this would be necessary

```go
func SkipDestructive(text, remote string) (skip bool) {
    var flag string
    switch {
       case fs.Config.DryRun:
            flag := "--dry-run"
            skip = true
       case fs.Config.Interactive:
            flag = "--interactive"
            // ask user here
            interactiveMu.Lock()
            defer interactiveMu.Unlock()
            // use config.Choose maybe
        default:
            skip = false
    }
    if skip {
        fs.Logf(src, "Not %s as %s", o, action, flag)
    }
    
}
```

**Comment by jeff-geoff** [comment]
Would it be easier to prompt once at the beginning and say eg “rclone purge will delete 850 files and 10 directories. Are you sure? Y/n”

I realize that doesn’t mimick cp

**Comment by HowJMay** [comment]
May I try to take this issue?

**Comment by ncw** [comment]
That would be great. Let me know if you need help.

**Comment by fishbullet** [comment]
@ncw I'm trying this one, ok ?

**Comment by ncw** [comment]
> I'm trying this one, ok ?

Super - thanks :-) . Let me know if you need help.

**Comment by ncw** [comment]
I've merged this now - thank you @philidor-green for your work.

It will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.53
